app/controllers/api/v0x1/portfolios_controller.rb

```diff
@@ -43,6 +43,37 @@ def destroy
         end
       end
 
+      def share
+        portfolio = Portfolio.find(params.require(:portfolio_id))
+        options = {:app_name      => ENV['APP_NAME'],
+                   :resource_name => 'portfolios',
+                   :resource_ids  => [portfolio.id.to_s],
+                   :permissions   => params.require(:permissions),
+                   :group_uuids   => params.require(:group_uuids)}
+        RBAC::ShareResource.new(options).process
```

Comment:
If we want to keep the controllers thin it might make sense to move this processing into a service class which in turn class the RBAC service class.  

Does this qualify as enough to move out into a separate class?  @syncrou Thoughts?

Reply:
@gmcculloug Yes, I agree. It would be much better to see a service that takes the params as an argument and returns a service class and/or json to be rendered.

Reply:
@syncrou @gmcculloug 
It is following the service model approach it just creates an instance and calls process on it.
It is like a Service Class except it doesn't have a service name.